cosmwasm/register-p2p-tokens.ts

```diff
@@ -0,0 +1,141 @@
+import { Command, Option } from 'commander';
+import fs from 'fs';
+
+import { addEnvOption, printError } from '../common';
+import { ConfigManager } from '../common/config';
+import { validateParameters } from '../common/utils';
+import { isConsensusChain } from '../evm/utils';
+import { TokenData, registerToken } from './its';
+import { ClientManager, mainProcessor } from './processor';
+
+export type SquidTokenData = {
+    axelarChainId: string;
+    tokenAddress: string;
+};
+
+export type SquidToken = {
+    tokenId: string;
+    decimals: number;
+    tokenType: 'interchain' | 'customInterchain' | 'canonical';
+    chains: SquidTokenData[];
+    originAxelarChainId: string;
+};
+
+export type SquidTokens = {
+    [tokenId: string]: SquidToken;
+};
+
+export type SquidTokenInfoFile = {
+    tokens: SquidTokens;
+};
+
+async function forEachTokenInFile(
+    config: ConfigManager,
+    options,
+    processToken: (token: SquidToken, chain: SquidTokenData) => Promise<void>,
+) {
+    const { env, tokenIds, chains } = options;
+    const tokenIdsToProcess = new Set(tokenIds);
+    const chainsToProcess = new Set(chains);
+    const tokenInfoString = fs.readFileSync(`axelar-chains-config/info/tokens-p2p/tokens-${env}.json`, 'utf8');
+    const tokenInfo = JSON.parse(tokenInfoString) as SquidTokenInfoFile;
+
+    const filteredTokens: SquidToken[] = Object.values(tokenInfo.tokens).filter(
+        (token: SquidToken) => (tokenIds ? tokenIdsToProcess.has(token.tokenId) : true) && token.tokenType === 'interchain',
+    );
+
+    for (const token of filteredTokens) {
+        const filteredChains: SquidTokenData[] = token.chains.filter((chain: SquidTokenData) => {
+            try {
+                return (
+                    (chains ? chainsToProcess.has(chain.axelarChainId.toLowerCase()) : true) &&
+                    isConsensusChain(config.getChainConfig(chain.axelarChainId.toLowerCase()))
+                );
+            } catch (e) {
+                printError(`Error getting chain config for ${chain.axelarChainId} (skipping chain): ${e}`);
+                return false;
+            }
+        });
+
+        for (const chain of filteredChains) {
+            await processToken(token, chain);
```

Comment:
catch the error here, print, and continue on error. return a bool to indicate success or if there was an error. Calling code can fail at the end if it returns an error